### PR TITLE
Take into acount send permission (power level)

### LINKED ISSUE #602: Hide message editor when user's power level is not sufficient to send message.
This message is displayed on Element Web:

image

Currently it's possible to type and send message, the local echo is displayed, but the message is not sent (rejected by the HS), and no error is displayed.

### PR DESCRIPTION
**Pull request by bmarty**
Fixes #602

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt [hunk 1]
```diff
@@ -218,4 +222,13 @@ class MessagesPresenter @Inject constructor(
             MessageComposerEvents.SetMode(composerMode)
         )
     }
+
+    @Composable
+    private fun getCanSendEvent(type: MessageEventType): State<Boolean> {
+        val canSendEvent = remember(type) { mutableStateOf(false) }
```
Comment (ganfra) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:228]: I guess we could add this method to matrix-ui like [this](https://github.com/vector-im/element-x-android/blob/6497a3f86ceaa36c9e0c0571bea80e77fcacf221/libraries/matrixui/src/main/kotlin/io/element/android/libraries/matrix/ui/room/MatrixRoomMembers.kt).

Also you should probably use `produceState` method (which does the same internally)

```
return produceState(initialValue = false, key1 = type){
        room.canSendEvent(type).getOrElse { false }
 }
```

**Review by ganfra — APPROVED** [review]
One remark otherwise LGTM

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt (hunk 1 above)
Comment (bmarty) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt]: Good point, thanks! a8fbcb0b1676d7a447037a68b284c2ea8c9a18e9

**Comment by ganfra** [comment]
Thanks, lets merge as soon as CI is happy!

**Pull request merged**